numpy/core/src/umath/string_ufuncs.cpp

```diff
@@ -185,6 +221,34 @@ string_comparison_loop(PyArrayMethod_Context *context,
 }
 
 
+
+template <bool rstrip, typename character>
+static void
+string_isalpha_loop(char **args, npy_intp const *dimensions,
+                    npy_intp const *steps, void *NPY_UNUSED(extra))
+{
+    /*
+     * Note, fetching `elsize` from the descriptor is OK even without the GIL,
+     * however it may be that this should be moved into `auxdata` eventually,
+     * which may also be slightly faster/cleaner (but more involved).
+     */
+    int len = steps[0] / sizeof(character);
```

Comment:
I had ignored it, assuming this was a first iteration for testing.  Yes, I know it adds some boilerplate, but we should probably just create the ufunc without any loops and then add them the same way as the string equality is done.

Then change the signature of this loop and the above comment actually can make sense.

Right now, you should get crashes for some sliced arrays `arr = arr[::10]`

Reply:
@seberg I think I've got something better now. Can you check again and let me know whether I've missed anything else?